src/Swift.Runtime/src/Swift/Data.cs

```diff
@@ -1,10 +1,119 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using Swift;
+using Swift.Runtime;
+
+
 /// <summary>
-/// Represents Foundation.Data type.
+/// Represents Swift Foundation.DataProtocol in C#.
+/// </summary>
+public interface ISwiftDataProtocol { }
+
+/// <summary>
+/// Represents Swift Foundation.ContiguousBytes in C#.
 /// </summary>
+public interface ISwiftContiguousBytes { }
 
-// TODO: https://github.com/dotnet/runtimelab/issues/2992
-public struct Data
+/// <summary>
+/// Represents Foundation.Data type.
+/// </summary>
+public struct Data : ISwiftObject
 {
     private long _flags;
     private IntPtr _object;
+
+    private static nuint _payloadSize = SwiftObjectHelper<Data>.GetTypeMetadata().Size;
```

Comment:
Nit: dotnet/runtime naming convention is to prefix static fields with `s_` - in case you would like to be consistent with it.

Reply:
Thanks, we will update the generated and manual bindings.